### DISCUSSION TITLE
FogVolume Diffucult shape

### DISCUSSION THREAD

**Discussion by dmitrii1991** [godotengine/godot-proposals#11302] [3D]
The ability to set the shape of the FogVolume ArrayMesh grid over which volume the fog will spread.

The problem is that I am writing a tactical game, I need to describe the fog on a group of hexagons - and to produce zones with fog content and absence is very expensive for empty areas

**Comment by Calinou** [comment]
This can already be done by providing a custom FogVolume density texture, which can be modified at runtime. [There is no Image3D class for this](https://github.com/godotengine/godot-proposals/issues/9583) though, so this requires more code than it could currently.

The FogVolume must cover the entire area that may receive fog for this to work, but the cost of volumetric fog rendering will remain mostly the same as if no density texture was used.
